packages/vscode-extension/src/project/project.ts

```diff
@@ -585,17 +585,22 @@ export class Project
     }
   };
 
-  private checkIfNativeChanged = throttle(async () => {
-    if (!this.detectedFingerprintChange && this.projectState.selectedDevice) {
-      if (await didFingerprintChange(this.projectState.selectedDevice.platform)) {
-        this.detectedFingerprintChange = true;
+  private checkIfNativeChanged = throttle(async (cancelToken: CancelToken) => {
+    if (!this.buildCacheInvalidated && this.projectState.selectedDevice) {
+      // TODO: should be refactored to not create a new PlatformBuildCache
+      // instance every time
+      const platform = this.projectState.selectedDevice.platform;
+      const buildCache = new PlatformBuildCache(platform, cancelToken);
+
+      if (await buildCache.isCacheInvalidated()) {
+        this.buildCacheInvalidated = true;
         this.eventEmitter.emit("needsNativeRebuild");
       }
     }
   }, 300);
```

Comment:
Build cache shouldn't live here but I considered few options and all of them made structure more complex.